### PR TITLE
SERVICES-1006 improve asset history elastic query

### PR DESCRIPTION
**Pull request by johnykes**

### CODE REVIEW

**Review by danielailie — CHANGES_REQUESTED** [review]

File: src/modules/asset-history/assets-history.service.ts [hunk 1]
```diff
@@ -73,15 +112,15 @@ export class AssetsHistoryService {
       }
     }
```
[the remark this answers is not in the document]
Comment (johnykes) [reply src/modules/asset-history/assets-history.service.ts:113]: Done

File: src/modules/asset-history/assets-history.service.ts [hunk 2]
```diff
@@ -50,20 +56,53 @@ export class AssetsHistoryService {
     collection: string,
     nonce: string,
     limit: number,
-    timestamp: string | number,
+    beforeTimestamp: number,
     historyLog: AssetHistoryLog[] = [],
   ): Promise<AssetHistoryLog[]> {
```
[the remark this answers is not in the document]
Comment (johnykes) [reply src/modules/asset-history/assets-history.service.ts:61]: Done

**Review by danielailie — APPROVED** [review]

File: src/modules/asset-history/assets-history.service.ts [hunk 3]
```diff
@@ -50,20 +56,54 @@ export class AssetsHistoryService {
...
     let elasticLogs = [];
-    let totalHits = 0;
-    let elasticTimestamp = 0;
+    const encodedCollection = Buffer.from(collection).toString('base64');
+    const encodedNonce = Buffer.from(nonce, 'hex').toString('base64');
+
```
[the remark this answers is not in the document]
Comment (johnykes) [reply src/modules/asset-history/assets-history.service.ts]: Fixed.

File: src/modules/asset-history/assets-history-elastic.service.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  ElasticSortOrder,
+  QueryType,
+  RangeLowerThan,
+} from '@elrondnetwork/erdnest';
+import { constants } from 'src/config';
+
+@Injectable()
+export class AssetsHistoryElasticService {
+  constructor(
+    private readonly elrondElasticService: ElrondElasticService, // @Inject(forwardRef(() => AssetsHistoryService)) // private readonly assetsHistoryService: AssetsHistoryService,
```
[the remark this answers is not in the document]
Comment (johnykes) [reply src/modules/asset-history/assets-history-elastic.service.ts]: Done.

**Review by danielailie — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Pull request merged**